src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkTableSize(size);
+        Node<K, V> node = new Node<>(key, value, null);
+        Node<K, V> tempNode = table[findIndex(key)];
+        if (tempNode == null) {
+            table[findIndex(key)] = node;
```

Comment:
```suggestion
        Node<K, V> tempNode = table[findIndex(key)];
        if (tempNode == null) {
            table[findIndex(key)] = node;
```
dont call getIndex() twice - create variable for this value